python/tank/platform/software_launcher.py

```diff
@@ -336,47 +382,186 @@ def get_standard_plugin_environment(self):
-        and returns a boolean to indicate whether it is supported by the launcher
-        or not.
+        Performs a search for supported software installations.
+
+        This method can be overridden by a subclass to completely override
+        supported software discovery. The typical requirement, however, is for
+        a subclasses to simply implement the ``_scan_software()`` abstract
+        method to return all variations of the software on the local filesystem.
 
-        :param str version: Version string to test, e.g. "2015" or "2017.3.sp4"
-        :returns: True if supported, False otherwise.
+        The default implementation calls ``_scan_software()`` to return
+        a list of ``SoftwareVersion`` objects, then calls ``_is_supported()`` on
+        each to determine the final list of supported executables.
         """
-        if self.minimum_supported_version is None:
-            return True
 
-        if is_version_older(version, self.minimum_supported_version):
+        software_versions = []
+
+        # retrieve a list of SoftwareVersion objects for all executables
+        for sw_version in self._scan_software():
+            (supported, reason) = self._is_supported(sw_version)
+            if supported:
+                self.logger.debug("Accepting %s" % (sw_version,))
+                software_versions.append(sw_version)
+            else:
+                self.logger.debug(
+                    "Rejecting %s. Reason: %s" % (sw_version, reason))
+                continue
+
+        return software_versions
+
+    ##########################################################################################
+    # protected methods
+
+    def _is_supported(self, sw_version):
+        """
+        Inspects the supplied :class:`SoftwareVersion` object to see if it
+        aligns with this launcher's known product and version limitations. Will
+        check the :meth:`~minimum_supported_version` as well as the list of
+        product and version filters.
+
+        :param sw_version: :class:`SoftwareVersion` object to test against the
+            launcher's product and version limitations.
+
+        :returns: A tuple of the form: ``(bool, str)`` where the first item
+            is a boolean indicating whether the supplied ``SoftwareVersion`` is
+            supported or not. The second argument is ``""`` if supported, but if
+            not supported will be a string representing the reason the support
+            check failed.
+
+        This method can be used by subclasses that override the
+        ``get_supported_software`` method.
+
+        The method can be overridden by subclasses that require more
+        sophisticated ``SoftwareVersion`` support checks. Alternatively, the
+        ``get_supported_software`` can implement its own support checks using
+        the more specific convenience methods ``_is_version_supported`` and
```

Comment:
Maybe add a blur that mentions that if both method return `True`then this method will return `True`as well, so that is clear that there is not an undocumented side-effect that would make the method return `False`.